src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,127 @@
+            table[indexForValue] = newNode;
+            size++;
+        } else {
+            isUniqueKey(newNode, table);
+        }
+        if (size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int indexWantedValue = indexInTable(key);
+        if (table[indexWantedValue] == null) {
+            return null;
+        }
+        if (Objects.equals(table[indexWantedValue].key, key)) {
+            return table[indexWantedValue].value;
+        } else {
+            Node<K, V> wantedNode = table[indexWantedValue];
+            while (wantedNode != null) {
+                if (isEquals(wantedNode.key, key)) {
+                    return wantedNode.value;
+                }
+                wantedNode = wantedNode.next;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
+        int hash;
+        K key;
+        V value;
+        Node<K,V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private void resize() {
+        int newThreshold = threshold * CAPACITY_AND_THRESHOLD_MULTIPLIER;
+        int newCapacity = capacity * CAPACITY_AND_THRESHOLD_MULTIPLIER;
+        Node<K, V>[] oldTable = table;
+        table = (Node<K, V>[]) new Node[newCapacity];
+        threshold = newThreshold;
+        int indexInNewTable;
+        for (Node<K, V> currentNode : oldTable) {
+            if (currentNode != null) {
+                indexInNewTable = indexInTable(currentNode.key);
+                if (table[indexInNewTable] == null) {
+                    table[indexInNewTable] = currentNode;
+                } else {
+                    isUniqueKey(currentNode, table);
+                }
+            }
+        }
+    }
+
+    private void isUniqueKey(Node<K, V> current, Node<K, V>[] table) {
+        int indexForValue = indexInTable(current.key);
+        Node<K, V> checkedNode = table[indexForValue];
+        if (isEquals(current.key, checkedNode.key)) {
+            checkedNode.value = current.value;
+        } else {
+            while (checkedNode.next != null) {
+                if (isEquals(current.key, checkedNode.key)) {
+                    checkedNode.value = current.value;
+                    return;
+                }
+                checkedNode = checkedNode.next;
+            }
+            if (isEquals(current.key, checkedNode.key)) {
+                checkedNode.value = current.value;
+                return;
+            }
+            checkedNode.next = current;
+            size++;
+        }
+    }
+
+    private boolean isEquals(K key, K secondKey) {
+        return key == null && secondKey == null || Objects.equals(key, secondKey);
```

Comment:
I see that this method is used to compare two objects with possible null value, but I think that default `key.equals(secondKey)` would be enough since `.equals()` should be provided by the class of the key. In such case, we also need to add `key != null` to the right side of the `||`
(same think with `Objects.equals` in the `getValue()` method)

Reply:
Need help from mentors. We have a question, which variant of code is correct?

Reply:
![image](https://user-images.githubusercontent.com/36365171/102241136-4d570d80-3f01-11eb-8fbe-a2f7f0ff3994.png)